src/poliastro/plotting/porkchop.py

```diff
@@ -0,0 +1,172 @@
+        return (
+            c3_launch.to(u.km ** 2 / u.s ** 2).value,
+            c3_arrival.to(u.km ** 2 / u.s ** 2).value,
+            tof.jd,
+        )
+
+    except AssertionError:
+        return None, None, None
+
+
+# numpy.vectorize is amazing
+targetting_vec = np.vectorize(
+    _targetting,
+    otypes=[np.ndarray, np.ndarray, np.ndarray],
+    excluded=["departure_body", "target_body"],
+)
+
+
+def porkchop(
+    departure_body, target_body, launch_span, arrival_span, ax=None, time_lines=True
+):
+    """Plots porkchop between two bodies.
+
+    Parameters
+    ----------
+    departure_body: poliastro.bodies.Body
+        Body from which departure is done
+    target_body: poliastro.bodies.Body
+        Body for targetting
+    launch_span: astropy.time.Time
+        Time span for launch
+    arrival_span: astropy.time.Time
+        Time span for arrival
+    ax: matplotlib.axes.Axes:
+        For custom figures
+    
+    Returns
+    -------
+    c3_launch: np.ndarray
+        Characteristic launch energy
+    c3_arrrival: np.ndarray
+        Characteristic arrival energy
+    tof: np.ndarray
+        Time of flight for each transfer
+
+    Example
+    -------
+    >>> from poliastro.plotting.porkchop import porkchop
+    >>> from poliastro.bodies import Earth, Mars
+    >>> from poliastro.util import time_range
+    >>> launch_span = time_range("2005-04-30", end="2005-10-07")
+    >>> arrival_span = time_range("2005-11-16", end="2006-12-21")
+    >>> c3dpt, c3arr, tof = porkchop(Earth, Mars, launch_span, arrival_span)
+    """
+
+    c3_launch, c3_arrival, tof = targetting_vec(
+        departure_body,
+        target_body,
+        launch_span[np.newaxis, :],
+        arrival_span[:, np.newaxis],
+    )
+
+    # Start drawing porkchop
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(15, 15))
+    else:
+        fig = ax.figure
+
+    c3_levels = np.linspace(0, 45, 30)
+
+    c = ax.contourf(
+        [D.to_datetime() for D in launch_span],
+        [A.to_datetime() for A in arrival_span],
+        c3_launch,
+        c3_levels,
+    )
+
+    l = ax.contour(
+        [D.to_datetime() for D in launch_span],
+        [A.to_datetime() for A in arrival_span],
+        c3_launch,
+        c3_levels,
+        colors="black",
+        linestyles="solid",
+    )
+
+    cbar = fig.colorbar(c)
+    cbar.set_label("km2 / s2")
+    plt.clabel(l, inline=1, fmt="%1.1f", colors="k", fontsize=10)
+
+    if time_lines:
+
+        time_levels = np.linspace(100, 500, 5)
+
+        t = ax.contour(
+            [D.to_datetime() for D in launch_span],
+            [A.to_datetime() for A in arrival_span],
+            tof,
+            time_levels,
+            colors="red",
+            linestyles="dashed",
+            linewidths=3.5,
+        )
+
+        ax.clabel(t, inline=1, fmt="%1.1f", colors="r", fontsize=14)
+
+    ax.grid()
+    fig.autofmt_xdate()
+
+    ax.set_title(
+        "{} - {} for year {}, C3 Launch, TFL".format(
```

Comment:
What does TFL mean?

Reply:
I found porkchops may include some extra information such us:

- TFL: Time of Flight
- DLA: Declination of the Launch Azimuth
- HCA. Heliocentric Angle (between the two boundary vectors)
- VHP: Arrival velocity at target body

We may allow the user to plot previous parameters, it could be interesting. Even if a spacecraft can fullfill de <a href="https://www.codecogs.com/eqnedit.php?latex=\Delta&space;v" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\Delta&space;v" title="\Delta v" /></a> requirements, DLA or others may compromise the mission. For example, I reat at `Orbital Mechanics and Astrodynamics (Gerald R.Hintz)` that launches to ISS require DLA about 50 deg.